wrappers/unity/Assets/RealSenseSDK2.0/Shaders/BGSeg.shader

```diff
@@ -0,0 +1,78 @@
+﻿// Upgrade NOTE: replaced 'mul(UNITY_MATRIX_MVP,*)' with 'UnityObjectToClipPos(*)'
+
+Shader "Custom/BGSeg" {
+	Properties
+	{
+		_MainTex("MainTex", 2D) = "white" {}
+		_ColorTex("Base (RGB)", 2D) = "white" {}
+        _MinRange("Min Range(m)", float) = 0.5
+        _MaxRange("Max Range(m)", float) = 2.0
+        _DepthScale("Depth Multiplyer Factor to Meters", float) = 1
+		[Toggle] _HasColor ("Has Color?", Float) = 0
+	}
+	SubShader
+	{
+		Tags { "QUEUE" = "Transparent" "IGNOREPROJECTOR" = "true" "RenderType" = "Transparent" "PreviewType" = "Plane" }
+		Pass 
+		{
+			ZWrite Off
+			Cull Off
+            Blend SrcAlpha OneMinusSrcAlpha
+			Fog { Mode Off }
+
+			CGPROGRAM
+			#pragma vertex vert_img
+			#pragma fragment frag
+			#pragma target 3.0
+			#pragma glsl
+			#pragma multi_compile _HASCOLOR_OFF _HASCOLOR_ON
+
+
+			#include "UnityCG.cginc"
+
+			sampler2D _MainTex;
+			sampler2D _ColorTex;
+			float _MinRange;
+            float _MaxRange;
+            float _DepthScale;
+
+			half4 frag(v2f_img pix) : SV_Target
+			{
+                float r = tex2D(_MainTex, pix.uv).r;
+                float distMeters = r * 65536 * _DepthScale;
```

Comment:
0xffff=65535, is `r` texture coordinate or depth value?

Reply:
r is the depth value, normalized to [0-1]. multiplying by 0xffff returns it to `unit16_t` and then multiplying by depth scale to get the distance in meters.

Reply:
I'll document this